cheatsheets/JWT_Cheat_Sheet.md

```diff
@@ -0,0 +1,227 @@
+ weak).
+
+## Objective
+
+This cheatsheet provides tips to prevent common security issues when using JSON Web Tokens (JWT).
+
+## Consideration about Using JWT
+
+Even if a JWT token is "easy" to use and allow to expose services (mostly REST style) in a stateless way, it's not the
+ solution that fits for all applications because it comes with some caveats, like for example the question of the
+ storage of the token (tackled in this cheatsheet) and others.
+
+If your application does not need to be fully stateless, you can consider using traditional session system provided by
+ all web frameworks and follow the advice from the dedicated [session management cheat sheet](Session_Management_Cheat_Sheet.md).
+ Especially for authenticating users, the use of tools like Oauth2 or OIDC backed by SAML has become an industry best
+ practice, as it allows one SAML implementation/Server to integrate a large number of authentication best practices.
+ Some of those tools use JWT's under the hood.
+
+## Issues
+
+### None Hashing Algorithm
+
+#### Symptom
+
+This attack, described [here](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/), occurs
+ when an attacker alters the token and changes the hashing algorithm to indicate, through the *none* keyword, that the
+ integrity of the token has already been verified. As explained in the link above *some libraries treated tokens signed
+ with the none algorithm as a valid token with a verified signature*, so an attacker can alter the token claims and
+ the modified token will still be trusted by the application.
+
+#### How to Prevent
+
+First, use a JWT library that is not exposed to this vulnerability, or use a library that allows you to specify which
+ algorithms are considered acceptable and do not include (or explicitly exclude) the `none` algorithm. For example in
+ [`pyjwt.decode`](https://pyjwt.readthedocs.io/en/stable/api.html#jwt.decode) (python) you can explicitly state which
+ signature(s) is acceptable. Most libraries have fixed this issue, but as a rule, if you're using a symmetric key you
+should explicitly set which libraries are allowed.
+
+Second (and not shown) is to use an asymmetric key to sign your jwts. In general, it should be seen as a best practice
+ to use an asymetrc key to sign jwts rather than a symmetric key.
+
+#### Implementation Example Symmetric Key
+
+=== "Java Example"
+    --8<-- "JWTCSA/0-verification.md:java"
+
+=== "Python Example (`pyjwt`)"
+    --8<-- "JWTCSA/0-verification.md:pyjwt"
+
+=== "JavaScript Example (`jose`)"
+    --8<-- "JWTCSA/0-verification.md:jose"
+
+### Token Sidejacking
+
+#### Symptom
+
+This attack occurs when a token has been intercepted/stolen by an attacker and they use it to gain access to the system
+ using targeted user identity. In part, this attack really can't be mitigated "in platform" as JWTs are generically stateless
```

Comment:
It could be mitigated using some proof-of-possession such as:

* sender-constrained JWT such as TLS-bound access tokens and DPoP access tokens;
* [SD-JWT bound to a public key pair](https://datatracker.ietf.org/doc/draft-ietf-oauth-selective-disclosure-jwt/)